### PR TITLE
expression: Support cast string as real push down (#25096)

### PR DESCRIPTION
**Pull request by ti-srebot**
cherry-pick #25096 to release-5.0
You can switch your code base to this Pull Request by using [git-extras](https://github.com/tj/git-extras):
```bash
# In tidb repo:
git pr https://github.com/pingcap/tidb/pull/25109
```

After apply modifications, you can push your change to this PR via:
```bash
git push git@github.com:ti-srebot/tidb.git pr/25109:release-5.0-e2340c1d8619
```

---

### What problem does this PR solve?

Issue Number: close #xxx 

Problem Summary:

### What is changed and how it works?

Proposal: [xxx](url) 

What's Changed:
- Support `CastStringAsReal` push down to TiFlash

How it Works:

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test

### Release note 

- No release note.

### CODE REVIEW

**Comment by leiysky** [comment]
Related to: https://github.com/pingcap/tics/issues/1936

**Review by leiysky — APPROVED** [review]
LGTM

**Comment by leiysky** [comment]
Related TiFlash issue: https://github.com/pingcap/tics/pull/2068

File: expression/expr_to_pb_test.go [hunk 1]
```diff
@@ -727,6 +727,11 @@ func (s *testEvaluatorSuite) TestExprPushDownToFlash(c *C) {
 	c.Assert(err, IsNil)
 	exprs = append(exprs, function)
 
+	// CastStringAsReal
+	function, err = NewFunction(mock.NewContext(), ast.Cast, types.NewFieldType(mysql.TypeString), realColumn)
```
Comment (LittleFall) [inline expression/expr_to_pb_test.go:731]: the third parameter is returnType, maybe you need to check the TiPB signature

**Review by LittleFall — APPROVED** [review]
others LGTM

**Pull request merged**